File: Platforms/DSSAT_ST_standalone/src/cs/DSSAT_ST_standalone/STEMP_State.cs
using System;
using System.Collections.Generic;
public class STEMP_State 
{
    private double[] _DSMID = new double[NL];
    private double[] _ST = new double[NL];
    private double _CUMDPT;
    private double[] _TMA = new double[NL];
    private double _SRFTEMP;
    private double _ATOT;
    
        public STEMP_State() { }
    
    
    public STEMP_State(STEMP_State toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    DSMID = new double[NL];
            for (int i = 0; i < NL; i++)
            { _DSMID[i] = toCopy._DSMID[i]; }
    
    ST = new double[NL];
            for (int i = 0; i < NL; i++)
            { _ST[i] = toCopy._ST[i]; }
    
    _CUMDPT = toCopy._CUMDPT;
    TMA = new double[NL];
            for (int i = 0; i < NL; i++)
            { _TMA[i] = toCopy._TMA[i]; }
    
    _SRFTEMP = toCopy._SRFTEMP;
    _ATOT = toCopy._ATOT;
    }
    }
    public double[] DSMID
        {
            get { return this._DSMID; }
            set { this._DSMID= value; } 
        }
    public double[] ST
        {
            get { return this._ST; }
            set { this._ST= value; } 
        }
    public double CUMDPT
        {
            get { return this._CUMDPT; }
            set { this._CUMDPT= value; } 
        }
    public double[] TMA
        {
            get { return this._TMA; }
            set { this._TMA= value; } 
        }
    public double SRFTEMP
        {
            get { return this._SRFTEMP; }
            set { this._SRFTEMP= value; } 
        }
    public double ATOT
        {
            get { return this._ATOT; }
            set { this._ATOT= value; } 
        }
}

File: Platforms/DSSAT_ST_standalone/src/java/DSSAT_ST_standalone/STEMP_State.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;
public class STEMP_State
{
    private Double [] DSMID;
    private Double [] ST;
    private Double CUMDPT;
    private Double [] TMA;
    private Double SRFTEMP;
    private Double ATOT;
    
    public STEMP_State() { }
    
    public STEMP_State(STEMP_State toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            DSMID = new Double[toCopy.getDSMID().length];
        for (int i = 0; i < toCopy.getDSMID().length; i++)
        {
            DSMID[i] = toCopy.getDSMID()[i];
        }
            ST = new Double[toCopy.getST().length];
        for (int i = 0; i < toCopy.getST().length; i++)
        {
            ST[i] = toCopy.getST()[i];
        }
            this.CUMDPT = toCopy.getCUMDPT();
            TMA = new Double[toCopy.getTMA().length];
        for (int i = 0; i < toCopy.getTMA().length; i++)
        {
            TMA[i] = toCopy.getTMA()[i];
        }
            this.SRFTEMP = toCopy.getSRFTEMP();
            this.ATOT = toCopy.getATOT();
        }
    }
    public Double [] getDSMID()
    { return DSMID; }

    public void setDSMID(Double [] _DSMID)
    { this.DSMID= _DSMID; } 
    
    public Double [] getST()
    { return ST; }

    public void setST(Double [] _ST)
    { this.ST= _ST; } 
    
    public Double getCUMDPT()
    { return CUMDPT; }

    public void setCUMDPT(Double _CUMDPT)
    { this.CUMDPT= _CUMDPT; } 
    
    public Double [] getTMA()
    { return TMA; }

    public void setTMA(Double [] _TMA)
    { this.TMA= _TMA; } 
    
    public Double getSRFTEMP()
    { return SRFTEMP; }

    public void setSRFTEMP(Double _SRFTEMP)
    { this.SRFTEMP= _SRFTEMP; } 
    
    public Double getATOT()
    { return ATOT; }

    public void setATOT(Double _ATOT)
    { this.ATOT= _ATOT; } 
    
}